text/0003-browser-support.md

```diff
@@ -0,0 +1,45 @@
+# Browser support
+- Start Date: 2019/04/26
+- RFC PR: https://github.com/openmrs/openmrs-rfc-frontend/pull/3
+
+## Decision, including impact on distributions
+Core OpenMRS frontend code supports the following browsers:
+- Last 2 Safari major versions
+- Last 3 Edge major versions
+- Last 5 Chrome major versions
+- Last 3 Firefox major versions
+- Last 3 Opera major versions
+
+The following browsers are not supported:
+- Internet Explorer
+
+Distributions may choose which browsers they support, including to drop support for browsers that they do not care about.
+
+## Definition
+Supporting a browser means ensuring that the javascript code meets the syntax implemented by that browser and does not use any
+DOM or javascript functionality that does not exist or is not implemented in that browser.
+
+Accomplishing this can be done with one of more of the following strategies:
+- Only write code that works all the supported browsers.
+- Write code that is not supported by all browsers, but compile it to older forms of javascript. The most popular options for doing
+this are [babel](https://babeljs.io/) and [typescript](https://www.typescriptlang.org/).
+- Use [polyfills](https://en.wikipedia.org/wiki/Polyfill_(programming)) to enhance older browsers such that they support some
+of the modern javascript features.
+
+At OpenMRS, we employ all three strategies, relying most heavily on compilation via Babel. The [@babel/preset-env](https://babeljs.io/docs/en/babel-preset-env)
+npm package allows you to choose which browsers you want to support and then will automatically make sure that your modern fancy code works
+in those browsers. It does this by using the [browserslist](https://github.com/browserslist/browserslist) and [compat-table](https://github.com/kangax/compat-table)
+npm packages.
+
+## Reason for decision
```

Comment:
Regarding, supporting older versions of other browsers (current - v2/v3), I would rather accept a bigger download (and hopefully cached so not to download again in defined time), because I am not sure of the assumption of "openmrs is used by people who can be instructed by modern browser"  in all contexts. Many places the browsers at access points, are seldom replaced for years, and auto-update may not work since in-LAN deployment, might not have internet.

Reply:
For the implementations that i have been directly involved with, part of the requirements has been to install a certain browser on each client. Could we do a post on talk and ask if there are implementations out there where this would work against them?

Reply:
Yeah thanks for sharing this angshu. Is IE11 support a thing that is a must for a lot of Bahmni users?

Reply:
For reference, here is info about end of life for IE10 and IE11 - https://en.wikipedia.org/wiki/Internet_Explorer_11. Microsoft has announced end of support for IE10 in January 2020, and IE11 will only receive critical security fixes as it becomes older and older.